castle/cms/browser/content/analytics.py

```diff
@@ -38,29 +44,41 @@ def ga_api_call(self, paths):
             result = None
 
         if result is None:
-            service = analytics.get_ga_service()
-            if not service:
-                return {'error': 'Could not get GA Service'}
-
-            profile = self.get_ga_profile(service)
-            if not profile:
-                return {'error': 'Could not get GA Profile'}
-
-            if self.request.get('type') == 'realtime':
-                ga = service.data().realtime()
-                if not params.pop('global', False):
-                    # need to restrict by filters
-                    path_query = ','.join(['rt:pagePath==%s' % p for p in paths])
-                    params['filters'] = path_query
+            form_type = self.request.form['type']
+            if '-ua' not in form_type:
+                ga_id = registry.get('castle.google_analytics_id', None)
+                service_key = registry.get('castle.google_api_service_key_file', None)
+
+                if ga_id:
+                    form = self.request.form
+                    # Set GOOGLE_ANALYTICS_IS_DEV env variable to true to use mock return data
+                    if os.environ.get("GOOGLE_ANALYTICS_IS_DEV", False):
+                        return get_mock_ga4_data(paths, form)
```

Comment:
a dev toggle like this should be moved into the get_ga4_data method itself so the code here doesn't have to concern itself with whether or not it is in a dev mode.